src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InProcOrRemoteHostAnalyzerRunner.cs

```diff
@@ -102,7 +103,22 @@ public InProcOrRemoteHostAnalyzerRunner(IDiagnosticAnalyzerService owner, Abstra
                 var analysisResult = await analyzerDriver.GetAnalysisResultAsync(cancellationToken).ConfigureAwait(false);
 
                 // if remote host is there, report performance data
-                if (client != null)
+                FireAndForgetReportAnalyzerPerformance(project, client, analysisResult, cancellationToken);
+
+                // get compiler result builder map
+                var builderMap = analysisResult.ToResultBuilderMap(project, version, analyzerDriver.Compilation, analyzerDriver.Analyzers, cancellationToken);
+
+                return DiagnosticAnalysisResultMap.Create(builderMap.ToImmutableDictionary(kv => kv.Key, kv => new DiagnosticAnalysisResult(kv.Value)), analysisResult.AnalyzerTelemetryInfo);
+            }
+
+            private async void FireAndForgetReportAnalyzerPerformance(Project project, RemoteHostClient client, AnalysisResult analysisResult, CancellationToken cancellationToken)
```

Comment:
❗️ Don't use `async void`. Here we can use `async Task` instead.

❕ Since the duration of this operation is not captured by the `Task` of the caller, it should create an `IAsyncToken` representing the operation.